Makefile

```diff
@@ -74,12 +74,48 @@ python: $(PYTHON_PATH)/bin/$(PYTHON_EXE)
 build: python
 	$(PYTHON_PATH)/bin/$(PYTHON_EXE) dev.py
 
+# copy to virtualenvs
+ROOT_FILES := $(HERE)/setup.py $(HERE)/setup.cfg $(HERE)/dev.py $(HERE)/README.rst $(HERE)/CHANGES.rst $(HERE)/buildout.cfg
+SRC_FILES := $(shell find $(HERE)/src ! -path '*egg-info*' \( -name '*.py' -o -name '*.txt' -o -name '*.test' \) )
+RCP_FILES := $(shell find $(HERE)/zc.recipe.egg_ ! -path '*egg-info*' \( -name '*.py' -o -name '*.txt' -o -name '*.rst' -o -name '*.cfg' -o -name '*.in' \) )
+DOC_FILES := $(shell find $(HERE)/doc -name '*.rst' -o -name '*.txt')
+
+ALL_COPY := $(subst $(HERE),$(VENV),$(SRC_FILES) $(DOC_FILES) $(RCP_FILES) $(ROOT_FILES))
+# Generate rules to map sources into targets
+$(foreach s,$(ALL_COPY),$(eval $s: $(VENV)/bin/$(PYTHON_EXE) $(subst $(VENV),$(HERE),$s)))
+
+$(ALL_COPY):
+	@test -d "$(dir $@)" || mkdir -p $(dir $@)
```

Comment:
`mkdir -p` silently does nothing if the directory already exists, so you could omit the `test -d ... ||` part.

Reply:
Yup